### PR TITLE
Rewrite donate page

### PR DESCRIPTION
**Pull request by torztomasz**

### CODE REVIEW

File: packages/frontend2/src/types/PropsFrom.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type PropsFrom<TComponent> = TComponent extends React.FC<infer Props>
+  ? Props
+  : never
```
Comment (torztomasz) [inline packages/frontend2/src/types/PropsFrom.ts:3]: I believe there was a built-in type, but I can not find it anywhere. Any idea?

Comment (torztomasz) [reply packages/frontend2/src/types/PropsFrom.ts:3]: yup

File: packages/frontend/src/pages/donate/view/DonatePage.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ function Header(props: HeaderProps) {
             transparency to the blockchain space.
           </p>
           <p className="mt-6">
-            We’re committed to delivering accurate and reliable information. We
+            We’re committed to delivering ac\rate and reliable information. We
```
Comment (imxeno) [inline packages/frontend/src/pages/donate/view/DonatePage.tsx:53]: A little oopsie happened here 😉

File: packages/frontend2/src/app/utils/getMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { type Metadata } from 'next'
+
+interface MetadataParams extends Metadata {
+  title: string
+  image: string
+  description?: string
+}
+
+export function getMetadata(params: MetadataParams): Metadata {
```
Comment (imxeno) [inline packages/frontend2/src/app/utils/getMetadata.ts:9]: Metadata is automatically merged in the app router, there's no need for this kind of stuff.

File: packages/frontend/src/pages/donate/view/DonatePage.tsx (hunk 2 above)
Comment (torztomasz) [reply packages/frontend/src/pages/donate/view/DonatePage.tsx:53]: good catch

File: packages/frontend2/src/app/utils/getMetadata.ts (hunk 3 above)
Comment (torztomasz) [reply packages/frontend2/src/app/utils/getMetadata.ts:9]: oh, forgot about that

**Review by imxeno — APPROVED** [review]

**Pull request merged**